src/www/ui/api/Helper/UserHelper.php

```diff
@@ -0,0 +1,144 @@
+
+      if ($ErrMsgs == null) {
+        $returnVal = new Info(200, "User updated succesfully!", InfoType::INFO);
+      } else {
+        $returnVal = new Info(400, $ErrMsgs, InfoType::INFO);
+      }
+    }
+    return $returnVal;
+  }
+
+  /**
+   * @param userDetails parsed from request body
+   * @return symfonyRequest $symfonyRequest
+   */
+  public function createSymRequest($userDetails)
+  {
+    $symfonyRequest = new \Symfony\Component\HttpFoundation\Request();
+    $symfonyRequest->request->set('user_pk', isset($userDetails['id']) ? $userDetails['id'] : $this->user_pk);
+    $symfonyRequest->request->set('user_name', isset($userDetails['name']) ? $userDetails['name'] : "");
+    $symfonyRequest->request->set('root_folder_fk', isset($userDetails['rootFolderId']) ? $userDetails['rootFolderId'] : 1);
+    $symfonyRequest->request->set('public', isset($userDetails['defaultVisibility']) ? $userDetails['defaultVisibility'] : "");
+    $symfonyRequest->request->set('default_folder_fk', isset($userDetails['defaultFolderId']) ? $userDetails['defaultFolderId'] : 1);
+    $symfonyRequest->request->set('user_desc', isset($userDetails['description']) ? $userDetails['description'] : "");
+    $symfonyRequest->request->set('_pass1', isset($userDetails['user_pass']) ? $userDetails['user_pass'] : "");
+    $symfonyRequest->request->set('_pass2', isset($userDetails['user_pass']) ? $userDetails['user_pass'] : "");
+    $symfonyRequest->request->set('_blank_pass', isset($userDetails['_blank_pass']) ? $userDetails['_blank_pass'] : "");
+    $symfonyRequest->request->set('user_status', isset($userDetails['user_status']) ? $userDetails['user_status'] : "");
+    $symfonyRequest->request->set('user_email', isset($userDetails['email']) ? $userDetails['email'] : "");
+    $symfonyRequest->request->set('email_notify', isset($userDetails['emailNotification']) && $userDetails['emailNotification'] ? "y" : "");
+    $symfonyRequest->request->set('default_bucketpool_fk', isset($userDetails['default_bucketpool_fk']) ? $userDetails['default_bucketpool_fk'] : 2);
+
+    if (isset($userDetails['accessLevel'])) {
+      $user_perm = 0;
+      switch ($userDetails['accessLevel']) {
+        case 'none':
+          $user_perm = Auth::PERM_NONE;
+          break;
+        case 'read_only':
+          $user_perm = Auth::PERM_READ;
+          break;
+        case 'read_write':
+          $user_perm = Auth::PERM_WRITE;
+        case 'clearing_admin':
+          $user_perm = Auth::PERM_CADMIN;
+        case 'admin':
+          $user_perm = Auth::PERM_ADMIN;
+        default:
+          break;
+      }
+      $symfonyRequest->request->set('user_perm', $user_perm);
+    } else {
+      $symfonyRequest->request->set('user_perm', 0);
+    }
+
+    // setting the checked agents
+    $agents = array();
+    if (isset($userDetails['agents'])) {
+      $agents['Check_agent_mimetype'] = isset($userDetails['agents']['mime']) && $userDetails['agents']['mime'] ? 1 : 0;
```

Comment:
```suggestion
      $agents['Check_agent_mimetype'] = isset($userDetails['agents']['mimetype']) && $userDetails['agents']['mimetype'] ? 1 : 0;
```

Reply:
sure will do this change!

Reply:
I think this will break existing `/jobs` POST API for creating jobs, as `agents` model is also being used there. As of now, do you suggest to change this, as more code changes needs to done to adapt this literal change?